velox/functions/lib/string/StringImpl.h

```diff
@@ -162,6 +162,95 @@ FOLLY_ALWAYS_INLINE int32_t charToCodePoint(const T& inputString) {
   return codePoint;
 }
 
+template <typename TInString>
+FOLLY_ALWAYS_INLINE int64_t
+hammingDistance(const TInString& left, const TInString& right) {
+  int64_t distance = 0;
+  int64_t leftPosition = 0;
+  int64_t rightPosition = 0;
+  std::string_view leftSV = std::string_view(left.data(), left.size());
+  std::string_view rightSV = std::string_view(right.data(), right.size());
+
+  while (leftPosition < leftSV.size() && rightPosition < rightSV.size()) {
+    int size = 0;
+    auto codePointLeft = utf8proc_codepoint(&leftSV.at(leftPosition), size);
+    leftPosition += size;
+    auto codePointRight = utf8proc_codepoint(&rightSV.at(rightPosition), size);
+    rightPosition += size;
+
+    if (codePointLeft != codePointRight) {
+      distance++;
+    }
+  }
+
+  VELOX_USER_CHECK(
+      leftPosition == leftSV.size() && rightPosition == rightSV.size(),
+      "The input strings to hamming_distance function must have the same length");
+
+  return distance;
+}
+
+FOLLY_ALWAYS_INLINE void castToCodePoints(
+    std::string_view str,
+    std::vector<int32_t>& codePoints) {
+  for (int index = 0; index < str.size();) {
+    int size = 0;
+    auto codePoint = utf8proc_codepoint(&str.at(index), size);
+    VELOX_USER_CHECK(
```

Comment:
I believe presto just ignores bad bytes and not throw an exception on encountering invalid utf characters - are you certain that presto also throws an exception (potentially killing job) on bad bytes ?

Reply:
[Presto does](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java#L631) for levenshtein, but not for hamming. I suspect the reason is that in hamming distance it is acceptable to ignore bad bytes and still arrive at an approximately correct answer - but levenshtein distance, which computes number of edits to convert one string to another, makes no sense with invalid bytes